app/master/cluster_master.py

```diff
@@ -173,11 +139,10 @@ def connect_slave(self, slave_url, num_executors, slave_session_id=None):
         # todo: Validate arg types for this and other methods called via API.
         # If a slave had previously been connected, and is now being reconnected, the cleanest way to resolve this
         # bookkeeping is for the master to forget about the previous slave instance and start with a fresh instance.
-        if slave_url in self._all_slaves_by_url:
-            old_slave = self._all_slaves_by_url.get(slave_url)
+        try:
```

Comment:
I find this less readable and less robust than it was previously. 

Less readable because with this large try catch it's not clear what line we're expecting to throw the ItemNotFoundError.
Less robust because if a line in [111, 125] unexpectedly throws an ItemNotFoundError, we'll silently drop the unexpected error.

I realize you were sort've forced into writing it like this because `get_slave` throws an error and you just copy pasted that code, but I don't like like that `get_slave` throws an error in the first place.

Reply:
Done.